qiskit/algorithms/gradients/param_shift_estimator_gradient.py

```diff
@@ -43,7 +41,24 @@ def __init__(self, estimator: BaseEstimator, options: Options | None = None):
                 default options > primitive's default setting.
                 Higher priority setting overrides lower priority setting
         """
-        self._gradient_circuits = {}
+        self._parameter_value_offset_cache = {}
+        self._SUPPORTED_GATES = [
+            "x",
+            "y",
+            "z",
+            "h",
+            "rx",
+            "ry",
+            "rz",
+            "p",
+            "cx",
+            "cy",
+            "cz",
+            "ryy",
+            "rxx",
+            "rzz",
+            "rzx",
+        ]
         super().__init__(estimator, options)
 
     def _run(
```

Comment:
This could be moved to the base class since this is the same for all gradients right? And then only `_run_unique` would need to be abstract 🙂

Reply:
But we don't want to apply `preprocess` and `postprocess` for the  finite diff gradient and SPSA gradient (In the future, we might have more numerical gradients and/or approximate gradients). Should we make `_run` like this?

```
def _run():
  if self._prepost_process:
     preprocess()
  
  run_unique
  
  if self._prepost_process:
    postprocess()
_
```